lib/em-http/http_connection.rb

```diff
@@ -190,6 +190,7 @@ def post_init
           # the connection because we're processing invalid HTTP
           @p.reset!
           unbind
+          :stop
```

Comment:
🗒️ This code was introduced in 3a9b96c2f31f29c112a11b934fee7d40cafd4dcc but the corresponding spec never worked. It failed with this error:

```
NoMethodError:
  undefined method `continue?' for nil:NilClass
```

This change fixes the failing spec as proposed by @jackorp in #356.